skypy/power_spectrum/_camb.py

```diff
@@ -72,29 +76,26 @@ def camb(wavenumber, redshift, cosmology, A_s, n_s):
                        omch2=cosmology.Odm0 * h2,
                        omk=cosmology.Ok0,
                        TCMB=cosmology.Tcmb0.value,
-                       mnu=np.sum(cosmology.m_nu.value),
+                       mnu=np.sum(cosmology.m_nu.to_value(u.eV)),
                        standard_neutrino_neff=cosmology.Neff
                        )
 
-    # camb requires redshifts to be in decreasing order
-    redshift_order = np.argsort(redshift)[::-1]
+    # camb interpolator requires redshifts to be in increasing order
+    redshift_order = np.argsort(redshift)
 
     pars.InitPower.ns = n_s
     pars.InitPower.As = A_s
 
-    pars.set_matter_power(redshifts=list(redshift[redshift_order]),
-                          kmax=np.max(wavenumber))
+    pars.Reion.redshift = z_reio
 
     pars.NonLinear = model.NonLinear_none
 
-    results = get_results(pars)
+    pk_interp = get_matter_power_interpolator(pars,
+                                              nonlinear=False,
+                                              hubble_units=False, k_hunit=False,
+                                              kmax=np.max(wavenumber),
+                                              zmax=np.max(redshift))
 
-    k = wavenumber * (1. / u.Mpc)
+    pzk = pk_interp.P(redshift[redshift_order], wavenumber)
```

Comment:
This is the matching change:
```suggestion
    pzk = pk_interp.P(redshift[redshift_order], wavenumber[wavenumber_order])
```